Fix scrolling when we change LazyListState

igordmn opened the pull request:
Fixes https://github.com/JetBrains/compose-multiplatform/issues/3551

When the new state is passed to LazyList, it goes down to `Modifier.pointerScrollable` as `controller: ScrollableState`, and wraps to `rememberUpdatedState`, which is always the same instance. Because it is the same instance, MouseWheelScrollableElement.update is never called.

Passing it is as a value now, as MouseWheelScrollableElement doesn't need a state.

## Testing

- Run SkikoScrollableTest
- Run the snippet from the issue

igordmn committed:
Fix scrolling when we change LazyListState

Fixes https://github.com/JetBrains/compose-multiplatform/issues/3551

When the new state is passed to LazyList, it goes down to `Modifier.pointerScrollable` as `controller: ScrollableState`, and wraps to `rememberUpdatedState`, which is always the same instance. Because it is the same instance, MouseWheelScrollableElement.update is never called.

Passing it is as a value now, as MouseWheelScrollableElement doesn't need a state.

eymar commented on compose/foundation/foundation/src/skikoTest/kotlin/androidx/compose/foundation/gestures/SkikoScrollableTest.kt:
```diff
@@ -0,0 +1,98 @@
+        var state by mutableStateOf(LazyListState())
+        setContent {
+            LazyColumn(state = state, modifier = Modifier.testTag("list").fillMaxSize()) {
+                items(1000) {
```
Sorry, I can't run the tests right now to check:
Does this test reproduce the original issue?
In the original issue it scrolls fine when 100 items added from the start. But it didn't work when first 10 added and then 100 added afterwards. 
And here the test adds 1000 from the beginning. 

Or the items count doesn't matter, because what matters is the LazyListState change?

igordmn replied:
Yes, it reproduces the orignal issue. It was not because we change items, but because we change the state.

I rechecked the original code - it is fixed.

Also, looking at the failed test now

igordmn committed:
Remember ScrollingLogic

It seems there is test that checks equality of the modifier across compositions. Probably for optimization.

Remembering scrolling logic seems logical here, and shouldn't add overhead.

In 1.6 it was fully rewritten to modifier node:

https://github.com/androidx/androidx/blob/a7e6bdf524be186baca08e9ddf7272b834a14bd2/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/Scrollable.kt#L276

igordmn commented on compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/Scrollable.kt:
```diff
@@ -283,13 +283,39 @@ private fun Modifier.pointerScrollable(
 private val NoOpOnDragStarted: suspend CoroutineScope.(startedPosition: Offset) -> Unit = {}
 
+@Composable
+private fun rememberScrollingLogic(
```
It seems there is test that checks equality of the modifier across compositions. Probably for optimization. `remember` fixes the test.

Remembering scrolling logic seems logical here, and shouldn't add overhead.

In 1.6 it was fully rewritten to modifier node:

https://github.com/androidx/androidx/blob/a7e6bdf524be186baca08e9ddf7272b834a14bd2/compose/foundation/foundation/src/commonMain/kotlin/androidx/compose/foundation/gestures/Scrollable.kt#L276

eymar replied:
which test was failing before adding `rememberScrollingLogic` ?

igordmn replied:
[This one](https://teamcity.jetbrains.com/buildConfiguration/JetBrainsPublicProjects_Compose_DesktopTestsForPullRequests/4287964?hideProblemsFromDependencies=false&hideTestsFromDependencies=false&expandBuildChangesSection=true&expandPull+Request+Details=true&expandBuildTestsSection=true)

eymar reviewed (approved):
LGTM. The tests passed on my machine too.
